Lib/subprocess.py

```diff
@@ -260,8 +260,29 @@ def _args_from_interpreter_flags():
         v = getattr(sys.flags, flag)
         if v > 0:
             args.append('-' + opt * v)
-    for opt in sys.warnoptions:
+
+    # -W options
+    warnoptions = sys.warnoptions
+    xoptions = getattr(sys, '_xoptions', {})
+    if 'dev' in xoptions and warnoptions and warnoptions[-1] == 'default':
+        # special case: -X dev adds 'default' to sys.warnoptions
+        warnoptions = warnoptions[:-1]
+    for opt in warnoptions:
         args.append('-W' + opt)
+
+    # -X options
+    if 'dev' in xoptions:
+        args.extend(('-X', 'dev'))
+    for opt in ('faulthandler', 'tracemalloc', 'importtime',
+                'showalloccount', 'showrefcount'):
+        if opt in xoptions:
+            value = xoptions[opt]
+            if value is True:
+                arg = opt
+            else:
+                arg = '%s=%s' % (opt, value)
```

Comment:
Do you want to use a pretty f-string here?

Reply:
Nope, I prefer to use str % args here :-) Don't ask me why, I have no rationale for such coding style choice :-)